test/test-signatures.spec.ts

```diff
@@ -86,6 +95,51 @@ describe("Signatures", function () {
       "R1A - asrt signed => error",
       testOneResponse("/invalid/response.root-unsigned.assertion-signed.xml", INVALID_SIGNATURE, 2)
     );
+    it(
+      "R1A - root signed - wantAssertionsSigned=true => error",
+      testOneResponse("/valid/response.root-signed.assertion-unsigned.xml", INVALID_SIGNATURE, 2, {
+        cert,
+        wantAssertionsSigned: true,
+      })
+    );
+    it(
+      "R1A - root signed - asrt unsigned encrypted -wantAssertionsSigned=true => error",
+      testOneResponse(
+        "/valid/response.root-signed.assertion-unsigned-encrypted.xml",
+        INVALID_ENCRYPTED_SIGNATURE,
+        2,
+        {
+          decryptionPvk: fs.readFileSync(__dirname + "/static/testshib encryption pvk.pem"),
+          cert,
+          wantAssertionsSigned: true,
+        }
+      )
+    );
+    it(
+      "R1A - root signed - asrt invalidly signed wantAssertionsSigned=true => error",
```

Comment:
@HendrikJan I tried to run this testcase in debugger in order to verify that it works as expeced i.e. that:
>5. signed response with signed unencrypted assertion so that assertions content (e.g. nameid) is modified after calculation of assertion's signature but prior to signing response (i.e. response's signature is valid but assertion's signature would not be)
> * expected result invalid signature

Purpose was that response's signature is valid and this code block at saml.ts:
https://github.com/node-saml/passport-saml/blob/bb0c32d0bf2532fdb706e5017495bff8674fdf91/src/passport-saml/saml.ts#L857
would set `validSignature = true` because top / msg level signature matched content.

Now - when `validSignature = true` due valid response level signature - assertion's signature check (because `wantAssertionsSigned = true`) should fail because `passport-saml` is instructed to check also assertion's signature even if top level signature was considered valid and overall result should be invalid signature exception.

Well...this test case ends up checking assertion's signature but due wrong reason (result would have been same even if `wantAssertionsSigned` would have been `false` because `passport-saml` tries to check assertions signature if saml responses signature was invalid).

Purpose of this test case was to to make sure/test that `passport-saml` checks assertions signature (when `wantAssertionsSigned = true`) EVEN if `validSignature` is already set to `true` due valid response level signature. In order to accomplish this test I suggested "validly signed saml reposponse which contains assertion with invalid assertion signature" because normally (when `wantAssertionsSigned = false`) such message would be considered valid (assertion's signature checking would be skipped).

It was quite painful exercise to setup mocha typescript testcase debugging env (without any proper js/ts toolchain background) so I might have done something incorrectly  but result (after stepping through `... mocha ... -g "R1A - root signed - asrt invalidly signed wantAssertionsSigned=true => error" ...`) seems to be that `invalid/response.root-signed.assertion-invalidly-signed.xml` is not constructed correctly because `passport-saml` considered top level signature to be invalid (it should have been valid so that test case works as expected). I assume `invalid/response.root-signed.assertion-invalidly-signed-encrypted.xml` has similar problem.

Reply:
Thank you so much for taking all this time and even setting up a debugging environment.

This PR has become much more involved than my original (very naive as I can see now) pull request.
Working with XML and signing and decrypting is new for me so I could easily have made a mistake.
I'll try to set up debugging later this week and step through the code.

Reply:
I've updated the response xml files.